version: '3.5'

services:
  grafana:
    image: grafana/grafana
    volumes:
      - grafana_data:/var/lib/grafana
      - ./../configs/grafana/provisioning/:/etc/grafana/provisioning/
      - ./../configs/grafana/grafana.ini:/etc/grafana/grafana.ini
    networks:
      - kaabah-network
    deploy:
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      labels:
        - "traefik.backend=grafana"
        - "traefik.enable=true"      
        - "traefik.docker.network=kaabah-network"
        - "traefik.port=3000"
        - "traefik.frontend.entryPoints=http,https"
        - "traefik.frontend.auth.basic=${AUTH_USER}:${AUTH_PASSWORD}"
        - "traefik.frontend.rule=Host:grafana.${SUBDOMAIN}"
      restart_policy:
        condition: on-failure

volumes:
  grafana_data: {}

networks:
  kaabah-network:
    name: ${DOCKER_NETWORK}
    external: true
    
